### PR TITLE
add eval method that isn't Core.eval

### PR DESCRIPTION
**Pull request by ZacLN**
Adds a binding for `eval` on entry to the top-level or a module scope. Are there other bindings that are automatically included in modules (other than themselves)?

### CODE REVIEW

File: src/scope.jl [hunk 1]
```diff
@@ -117,16 +117,19 @@ function scopes(x::EXPR, state)
     s0 = state.scope
     if headof(x) === :file
         setscope!(x, state.scope)
+        add_eval_method(x, state)
```
Comment (pfitzseb) [inline src/scope.jl:120]: Why do we need to add this here? Is this for basically `Main.eval`?

File: src/scope.jl [hunk 2]
```diff
@@ -137,4 +140,11 @@ function scopes(x::EXPR, state)
         end
     end
     return s0
-end
\ No newline at end of file
+end
+
+# Add an `eval` method
+function add_eval_method(x, state)
```
Comment (pfitzseb) [inline src/scope.jl:146]: I think that we should techinically add `include` here as well, at least for modules.

File: src/scope.jl (hunk 1 above)
Comment (ZacLN) [reply src/scope.jl:120]: Yeah it is. I don't fully understand how `Main` is managed (what is `MainInclude` for?)

File: src/scope.jl (hunk 2 above)
Comment (ZacLN) [reply src/scope.jl:146]: Thanks - you're right however I think I'll leave that one for the moment

**Review by pfitzseb — APPROVED** [review]

**Pull request merged**